Program.cs

```diff
@@ -1,4 +1,63 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// STRING PERMUTATIONS
+// Generate all string character permutations. 
+// For example, Input: "NOT", Output: "NOT NTO ONT OTN TNO TON" 
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>String permutations </summary>
+   internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>Getting input from the user </summary>
+      static void Main () {
+         Console.Write ("Enter the input:");
+         string input = Console.ReadLine ();
+         List<string> permutations = Permute (input);
+         Console.Write ("Output:");
+         //print each and every element in the list
+         foreach (string permutation in permutations) {
+           Console.WriteLine (permutation);
+         }
+      }
+      /// <summary>input permutations</summary>
+      /// <param name="str">input</param>
+      /// <returns>result</returns>
 
-   
\ No newline at end of file
+      static List<string> Permute (string str) {
+         List<string> result = new List<string> ();

```

Comment:
Make List of result as static variable in the program class. 
Here you are holding two references, one in permute method and another in Permute helper method.
Making it as static variable will reduce that cost.